avocado/utils/network/interfaces.py

```diff
@@ -338,6 +338,21 @@ def remove_ipaddr(self, ipaddr, netmask):
             msg = 'Failed to remove ipaddr. {}'.format(ex)
             raise NWException(msg)
 
+    def remove_link(self):
+        """Deletes virtual interface link.
+
+        This method will try to delete the virtual device link and the
+        interface will no more be listed with 'ip a' and if fails it
+        will raise a NWException. Be careful, you can lost connection.
+
+        You must have sudo permissions to run this method on a host.
+        """
+        cmd = "ip link del dev %s" % self.name
+        try:
+            run_command(cmd, self.host, sudo=True)
+        except Exception as ex:
+            raise NWException("Failed to delete link")
```

Comment:
Since you have the reason of the exception with `ex`, would be nice to show that information to the upper layers, inside the `NWException`'s message.

Reply:
Done